### ISSUE TITLE
[autocomplete-css] Support for SCSS files

### ISSUE THREAD

**Issue by savetheclocktower** [pulsar-edit/pulsar#542]
### Have you checked for existing feature requests?

- [X] Completed

### Summary

The `autocomplete-css` package has two big problems: (a) it doesn't seem to suggest things correctly when (legacy) Tree-sitter grammars are used, and (b) it doesn't have any code to handle SCSS syntax (as opposed to the old Sass syntax that had no braces).

The first item is low priority because we're about to phase out those grammars, and I'll make sure the modern Tree-sitter CSS grammar doesn't make the same mistake. But we should update all the Sass code paths so that they check for SCSS-specific scope names in addition to the Sass-specific scope names.

### What benefits does this feature provide?

CSS autocompletion will work inside of SCSS files.

### Any alternatives?

Don't think so. The whole reason there's a Sass-specific code path in the provider is that Sass blocks can be nested and CSS blocks can't. So the provider needs to know what mode it's in so that it knows whether it can suggest things like tag names.

**Comment by confused-Techie** [comment]
Are you aware if the completions SCSS files need would be any different? Since our COMPLETIONS.json file doesn't have anything that's SCSS specific, so I wonder if we would want to add those in, or if we are currently only providing value completions

**Comment by savetheclocktower** [comment]
I don't think they would be, but I did only glance at it. I think it could reuse the logic it uses for old-style Sass files.

**Comment by confused-Techie** [comment]
Do you think this was covered by #601 
Admittedly I didn't test SCSS much at all during that PR

**Comment by savetheclocktower** [comment]
No, didn't address it. The SCSS grammar is separate from the Sass grammar. There's a third-party tree-sitter library for SCSS, but it's not quite where it needs to be, so to make this work I'd have to spend some time with the builtin TM-style SCSS grammar.

**Comment by confused-Techie** [comment]
> No, didn't address it. The SCSS grammar is separate from the Sass grammar. There's a third-party tree-sitter library for SCSS, but it's not quite where it needs to be, so to make this work I'd have to spend some time with the builtin TM-style SCSS grammar.

Thanks a ton for clarifying, just wanted to double check so we could close it if need be. Didn't realize they were separated out

**Comment by savetheclocktower** [comment]
[`pulsar-ide-css`](https://packages.pulsar-edit.dev/packages/pulsar-ide-css) delivers autocompletion support for SCSS and Less files via the VS Code CSS language server, so that's one way we could go. #1378 has some of the earlier discussion about the idea of retiring `autocomplete-css` in favor of `pulsar-ide-css`.
